### PR TITLE
Move subscriptionId to global

### PR DESCRIPTION
**Pull request by vivsriaus**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sergey-shandar** [comment]
@vivsriaus is there a good reason to introduce SDK breaking changes in the old stable API version?

**Comment by sergey-shandar** [comment]
@vivsriaus the order of parameters does matter for some SDKs. Please, revert back to 
- the api version should be the first parameter,
- the subscription is should be the second parameter.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json [hunk 1]
```diff
@@ -44,14 +44,10 @@
         "description": "This operation provides all the locations that are available for resource providers; however, each resource provider may support a subset of this list.",
         "parameters": [
           {
-            "name": "subscriptionId",
-            "in": "path",
-            "required": true,
-            "type": "string",
-            "description": "The ID of the target subscription."
+            "$ref": "#/parameters/ApiVersionParameter"
           },
```
Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: @vivsriaus the order of parameters does matter for some SDKs. Please, revert back to
- the api version should be the first parameter,
- the subscription is should be the second parameter.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json (hunk 1 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: @sergey-shandar Fixed

**Comment by vivsriaus** [comment]
@sergey-shandar With regards to your comment about introducing breaking changes, we're doing this because our file was flagged as non compliant per Mohammed Zehgir. (non compliant because we were not using subId as a global parameter). Also, can you explain how this is considered a breaking change? The subscriptionId parameter is still required in the modified operations, right?

**Comment by sergey-shandar** [comment]
@vivsriaus it's breaking changes for **SDKs** because the parameters and their order are changed. We generate Azure SDK using the swagger files. So imagine Azure SDK C# DLL, one version of this DLL has a function with one parameter type and the new version of this DLL has a function with different parameter types and order of the parameters are swapped.

**Comment by sergey-shandar** [comment]
@vivsriaus thanks for change the order back. It's not an SDK breaking change anymore.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json [hunk 2]
```diff
@@ -323,6 +315,13 @@
     }
   },
   "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The ID of the target subscription."
+    },
```
Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: please, add `"x-ms-parameter-location": "client"`

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json [hunk 3]
```diff
@@ -323,6 +315,13 @@
...
     "ApiVersionParameter": {
       "name": "api-version",
       "in": "query",
```
Comment (sergey-shandar) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: please, add `"x-ms-parameter-location": "client"` to the `ApiVersionParameter` as well.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json (hunk 2 above)
Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: What does this give us?

Comment (vivsriaus) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2016-06-01/subscriptions.json]: Never mind, I get it now

**Review by sergey-shandar — APPROVED** [review]

**Comment by lmazuel** [comment]
@vivsriaus @sergey-shandar This is a massive breaking change :(. This is moving from a local parameter to a client parameter :(

**Comment by lmazuel** [comment]
But yeah, it's written client, so what the hell Python is generating out of this :(
............

**Comment by lmazuel** [comment]
@AutorestCI regenerate azure-sdk-for-python

**Comment by lmazuel** [comment]
Ok, this is definitely a massive breaking change....

**Pull request merged**